apis/marin3r/v1alpha1/resources.go

```diff
@@ -0,0 +1,237 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	"github.com/3scale-ops/marin3r/pkg/envoy"
+	envoy_serializer "github.com/3scale-ops/marin3r/pkg/envoy/serializer"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/yaml"
+)
+
+// Blueprint is an enum of the supported blueprints for
+// generated resources
+type Blueprint string
+
+const (
+	// TlsCertificate
+	TlsCertificate Blueprint = "tlsCertificate"
+	// TlsValidationContext
+	TlsValidationContext Blueprint = "tlsValidationContext"
+)
+
+// Resource holds serialized representation of an envoy
+// resource
+type Resource struct {
+	// Type is the type url for the protobuf message
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:validation:Enum=listener;route;scopedRoute;cluster;endpoint;secret;runtime;extensionConfig;
+	Type string `json:"type"`
+	// FromProto is the protobufer message that configures the resource. The proto
+	// must match the envoy configuration API v3 specification for the given resource
+	// type (https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#resource-types)
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Value *runtime.RawExtension `json:"value,omitempty"`
+	// The name of a Kubernetes Secret of type "kubernetes.io/tls"
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	GenerateFromTlsSecret *string `json:"generateFromTlsSecret,omitempty"`
+	// Specifies a label selector to watch for EndpointSlices that will
+	// be used to generate the endpoint resource
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	GenerateFromEndpointSlices *GenerateFromEndpointSlices `json:"generateFromEndpointSlices,omitempty"`
+	// Template specifies a template to generate a configuration proto. It is currently
```

Comment:
Is this description correct for `Blueprint`?

Reply:
Description is accurate, but the field name changed several times and I forgot to update the description accordingly. Fixed with 7a3d23e